src/core/pyproxy.js

```diff
@@ -227,20 +228,33 @@ JS_FILE(pyproxy_init_js, () => {0,0; /* Magic, see include_js_file.h */
      * Pyodide will automatically destroy the ``PyProxy`` when it is garbage
      * collected, however there is no guarantee that the finalizer will be run
      * in a timely manner so it is better to ``destory`` the proxy explicitly.
+     *
+     * @param {string} [destroyed_msg] The error message to print if use is
+     *        attempted after destroying. Defaults to "Object has already been
+     *        destroyed".
      */
-    destroy() {
+    destroy(destroyed_msg) {
       let ptrobj = _getPtr(this);
       Module.finalizationRegistry.unregister(this);
       // Maybe the destructor will call Javascript code that will somehow try
       // to use this proxy. Mark it deleted before decrementing reference count
       // just in case!
       this.$$.ptr = null;
+      this.$$.destroyed_msg = destroyed_msg;
       try {
         _Py_DecRef(ptrobj);
       } catch (e) {
         Module.fatal_error(e);
       }
     }
+    /**
+     * Make a new PyProxy pointing to the same Python object.
+     * Useful if someone else is destroying your PyProxy.
```

Comment:
"someone" makes mes wonder what other person might have access to those PyProxy.

Maybe "if your PyProxy gets destroyed elsewhere in the code" or something similar?

Reply:
Sounds good.